qiskit/providers/aer/noise/utils/idle_scheduler.py

```diff
@@ -0,0 +1,158 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Idle gates scheduler module
+
+The goal of this module is to add idle gates to a circuit instead of
+having blank spaces with no operations at times where no gate can be applied
+to the qubit (due to qubit dependenices or barriers)
+"""
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.converters import circuit_to_dag
+from qiskit.converters import dag_to_circuit
+from qiskit.extensions.standard.iden import IdGate
+
+
+def schedule_idle_gates(circuit, op_times=None, default_op_time=1, labels=None):
+    """
+    This function gets a circuit and returns a new one with the idle
+    gates in place; each idle gate is labeled "id_<name>" where <name> is the name of
+    the slowest gate that runs in parallel
+
+    Args:
+        circuit (QuantumCircuit): The circuit to add idle gates to
+        op_times (dictionary): dictionary of times taken by specific gate types
```

Comment:
Can this be changed to `gate_times (list)` to match `basic_device_noise_model` argument. And if in a layer all gates have time set to 0 no id should be inserted